src/4.3_to_5.0/migrateSettingsFolder.ts

```diff
@@ -38,6 +39,43 @@ function migrateSettingsMap(legacySettingsMap: {
     migratedSettingsMap["userFilters.areEnabled"] = areUserFiltersEnabled;
   }
 
+  const legacyDrillthroughColumns:
+    | {
+        [serverUrl: string]: {
+          [cubeName: string]: {
+            functionName: "Value" | "Caption" | "MemberValue" | "MemberCaption";
+            columnName: string;
+          }[];
+        };
+      }
+    | undefined =
+    legacySettingsMap["widgets.Tabular.drillthrough.selectedColumns"];
+
+  if (legacyDrillthroughColumns !== undefined) {
+    const migratedDrillthroughColumns: Settings["drillthrough.defaultSelectedColumns"] =
+      {};
+    // The legacy settings are per server and per cube.
+    // The new ones are only per cube: the assumption is that no customer has multiple servers containing a cube with the same name but exposing a different data model.
+    for (const serverUrl in legacyDrillthroughColumns) {
+      for (const cubeName in legacyDrillthroughColumns[serverUrl]) {
+        if (migratedDrillthroughColumns[cubeName] === undefined) {
+          migratedDrillthroughColumns[cubeName] = [];
+        }
+        migratedDrillthroughColumns[cubeName].push(
+          ...legacyDrillthroughColumns[serverUrl][cubeName].map(
+            ({ columnName }) => columnName,
+          ),
+        );
+      }
+    }
+
+    // Remove duplicates.
+    migratedSettingsMap["drillthrough.defaultSelectedColumns"] = _mapValues(
+      migratedDrillthroughColumns,
+      _uniq,
+    );
+  }
```

Comment:
This implies that two drillthrough columns with the same `columnName`  but different `functionName` are considered duplicates. Right ?

Reply:
Indeed. In fact there are even two ways to have duplicates:
- because UI4 was typically storing the following for each column:
```
{
  functionName: "MemberValue",
  columnName: "myColumn",
}, {
  functionName: "MemberCaption",
  columnName: "myColumn"
}
```
So we don't want to save each column twice in the new Atoti UI 5 setting.

The other way is more rare. It is because UI4 was saving the columns per server and per cube whereas UI5 only stores them per cube. So you could have a duplicate because a column was save for a cube with the same name under two different servers.

The current implementation in this PR should get rid of any duplicate, whether they come from the first way or the second one I described here.